File: src/main/java/com/bajdcc/LALR1/interpret/os/irq/IRPrint.java
package com.bajdcc.LALR1.interpret.os.irq;

import com.bajdcc.LALR1.interpret.os.IOSCodePage;
import com.bajdcc.util.ResourceLoader;

/**
 * 【中断】文字输出
 *
 * @author bajdcc
 */
public class IRPrint implements IOSCodePage {
	@Override
	public String getName() {
		return "/irq/print";
	}

	@Override
	public String getCode() {
		return ResourceLoader.INSTANCE.load(getClass());
	}
}

File: src/main/kotlin/com/bajdcc/LALR1/interpret/os/irq/IRPrint.kt
package com.bajdcc.LALR1.interpret.os.irq

import com.bajdcc.LALR1.interpret.os.IOSCodePage
import com.bajdcc.util.ResourceLoader

/**
 * 【中断】文字输出
 *
 * @author bajdcc
 */
class IRPrint : IOSCodePage {
    override val name: String
        get() = "/irq/print"

    override val code: String
        get() = ResourceLoader.load(javaClass)
}
